diff -Naur --no-dereference a/lib/Smokeping/probes/SSH.pm b/lib/Smokeping/probes/SSH.pm
see https://github.com/oetiker/SmokePing/commit/62ac9fda04b994bbf4f97d3dd1cf8b92cf279e71
--- a/lib/Smokeping/probes/SSH.pm	2020-06-27 14:54:05.705088092 +0200
+++ b/lib/Smokeping/probes/SSH.pm	2020-06-27 14:54:48.653105297 +0200
@@ -55,7 +55,7 @@
     # no need for this if we run as a cgi
     unless ( $ENV{SERVER_SOFTWARE} ) {
         
-        my $call = "$self->{properties}{binary} -t dsa,rsa,ecdsa $self->{properties}{init_host}";
+        my $call = "$self->{properties}{binary} -t rsa,ecdsa $self->{properties}{init_host}";
         my $return = `$call 2>&1`;
         if ($return =~ m/$ssh_re/s){
             print "### parsing ssh-keyscan output...OK\n";

