diff -Naur --no-dereference default-config/certs/ca.cnf config/certs/ca.cnf
--- default-config/certs/ca.cnf	1970-01-01 01:00:01.000000000 +0100
+++ config/certs/ca.cnf	2020-04-18 05:22:29.380588812 +0200
@@ -14,12 +14,12 @@
 RANDFILE		= $dir/.rand
 name_opt		= ca_default
 cert_opt		= ca_default
-default_days		= 60
+default_days		= 3650
 default_crl_days	= 30
 default_md		= sha256
 preserve		= no
 policy			= policy_match
-crlDistributionPoints	= URI:http://www.example.org/example_ca.crl
+#crlDistributionPoints	= URI:http://www.example.org/example_ca.crl
 
 [ policy_match ]
 countryName		= match
@@ -46,17 +46,17 @@
 output_password		= whatever
 x509_extensions		= v3_ca
 
-[certificate_authority]
-countryName		= FR
-stateOrProvinceName	= Radius
-localityName		= Somewhere
-organizationName	= Example Inc.
-emailAddress		= admin@example.org
-commonName		= "Example Certificate Authority"
-
 [v3_ca]
 subjectKeyIdentifier	= hash
 authorityKeyIdentifier	= keyid:always,issuer:always
 basicConstraints	= critical,CA:true
-crlDistributionPoints	= URI:http://www.example.org/example_ca.crl
+#crlDistributionPoints	= URI:http://www.example.org/example_ca.crl
+
+[certificate_authority]
+#countryName		= FR
+#stateOrProvinceName	= Radius
+#localityName		= Somewhere
+#organizationName	= Example Inc.
+#emailAddress		= admin@example.org
+#commonName		= "Example Certificate Authority"
 
diff -Naur --no-dereference default-config/certs/client.cnf config/certs/client.cnf
--- default-config/certs/client.cnf	1970-01-01 01:00:01.000000000 +0100
+++ config/certs/client.cnf	2020-04-18 05:08:38.426117692 +0200
@@ -45,9 +45,9 @@
 output_password		= whatever
 
 [client]
-countryName		= FR
-stateOrProvinceName	= Radius
-localityName		= Somewhere
-organizationName	= Example Inc.
-emailAddress		= user@example.org
-commonName		= user@example.org
+#countryName		= FR
+#stateOrProvinceName	= Radius
+#localityName		= Somewhere
+#organizationName	= Example Inc.
+#emailAddress		= user@example.org
+#commonName		= user@example.org
diff -Naur --no-dereference default-config/certs/inner-server.cnf config/certs/inner-server.cnf
--- default-config/certs/inner-server.cnf	1970-01-01 01:00:01.000000000 +0100
+++ config/certs/inner-server.cnf	2020-04-18 05:07:09.802200367 +0200
@@ -45,10 +45,10 @@
 output_password		= whatever
 
 [server]
-countryName		= FR
-stateOrProvinceName	= Radius
-localityName		= Somewhere
-organizationName	= Example Inc.
-emailAddress		= admin@example.org
-commonName		= "Example Inner Server Certificate"
+#countryName		= FR
+#stateOrProvinceName	= Radius
+#localityName		= Somewhere
+#organizationName	= Example Inc.
+#emailAddress		= admin@example.org
+#commonName		= "Example Inner Server Certificate"
 
diff -Naur --no-dereference default-config/certs/passwords.mk config/certs/passwords.mk
--- default-config/certs/passwords.mk	1970-01-01 01:00:01.000000000 +0100
+++ config/certs/passwords.mk	2020-04-17 02:24:47.314560249 +0200
@@ -3,4 +3,4 @@
 PASSWORD_CA	= 'whatever'
 PASSWORD_CLIENT	= 'whatever'
 USER_NAME	= 'user@example.org'
-CA_DEFAULT_DAYS  = '60'
+CA_DEFAULT_DAYS  = '3650'
diff -Naur --no-dereference default-config/certs/server.cnf config/certs/server.cnf
--- default-config/certs/server.cnf	1970-01-01 01:00:01.000000000 +0100
+++ config/certs/server.cnf	2020-04-18 05:05:28.652305902 +0200
@@ -14,7 +14,7 @@
 RANDFILE		= $dir/.rand
 name_opt		= ca_default
 cert_opt		= ca_default
-default_days		= 60
+default_days		= 3650
 default_crl_days	= 30
 default_md		= sha256
 preserve		= no
@@ -45,10 +45,10 @@
 output_password		= whatever
 
 [server]
-countryName		= FR
-stateOrProvinceName	= Radius
-localityName		= Somewhere
-organizationName	= Example Inc.
-emailAddress		= admin@example.org
-commonName		= "Example Server Certificate"
+#countryName		= FR
+#stateOrProvinceName	= Radius
+#localityName		= Somewhere
+#organizationName	= Example Inc.
+#emailAddress		= admin@example.org
+#commonName		= "Example Server Certificate"
 
diff -Naur --no-dereference default-config/clients.conf config/clients.conf
--- default-config/clients.conf	1970-01-01 01:00:01.000000000 +0100
+++ config/clients.conf	2020-04-17 03:59:52.220171206 +0200
@@ -97,7 +97,7 @@
 	#  The default secret below is only for testing, and should
 	#  not be used in any real environment.
 	#
-	secret = testing123
+	$INCLUDE ./client-secret.conf
 
 	#
 	#  Old-style clients do not send a Message-Authenticator
@@ -161,7 +161,7 @@
 	#  This is done by setting the "virtual_server" configuration
 	#  item, as in the example below.
 	#
-#	virtual_server = home1
+#	virtual_server = inner-tunnel
 
 	#
 	#  A pointer to the "home_server_pool" OR a "home_server"
diff -Naur --no-dereference default-config/mods-available/eap config/mods-available/eap
--- default-config/mods-available/eap	1970-01-01 01:00:01.000000000 +0100
+++ config/mods-available/eap	2020-04-19 05:21:01.883839504 +0200
@@ -321,16 +321,16 @@
 		#     'c_rehash' is OpenSSL's command.
 		#  3) uncomment the lines below.
 		#  5) Restart radiusd
-	#	check_crl = yes
+		check_crl = yes
 
 		# Check if intermediate CAs have been revoked.
 	#	check_all_crl = yes
 
-		ca_path = ${cadir}
+		ca_path = ${cadir}/ca_dir
 
 		# Accept an expired Certificate Revocation List
 		#
-	#	allow_expired_crl = no
+		allow_expired_crl = no
 
 		#  If check_cert_issuer is set, the value will
 		#  be checked against the DN of the issuer in
@@ -360,7 +360,10 @@
 		#  TLS-Client-Cert-Common-Name attribute.  This check
 		#  can be done via any mechanism you choose.
 		#
-	#	check_cert_cn = %{User-Name}
+		#FIXME Accounting uses the user name so the user shouldn't
+		#      be able to choose this. However, I would rather use
+		#      the CN for accounting and permit any user name.
+		check_cert_cn = %{User-Name}
 
 		#  Set this option to specify the allowed
 		#  TLS cipher suites.  The format is listed
@@ -459,7 +462,7 @@
 			#  If "enable = no" below, you CANNOT enable resumption for just one
 			#  user by setting the above attribute to "yes".
 			#
-			enable = no
+			enable = yes
 
 			#  Lifetime of the cached entries, in hours. The sessions will be
 			#  deleted/invalidated after this time.
@@ -474,7 +477,7 @@
 			#  set the "name" if you want to persist sessions (see
 			#  below).
 			#
-		#	name = "EAP module"
+			name = "EAP module"
 
 			#  Simple directory-based storage of sessions.
 			#  Two files per session will be written, the SSL
@@ -494,7 +497,7 @@
 			#
 			#  This feature REQUIRES "name" option be set above.
 			#
-		#	persist_dir = "${logdir}/tlscache"
+			persist_dir = "${db_dir}/tlscache"
 		}
 
 		#  As of version 2.1.10, client certificates can be
diff -Naur --no-dereference default-config/mods-available/sql config/mods-available/sql
--- default-config/mods-available/sql	1970-01-01 01:00:01.000000000 +0100
+++ config/mods-available/sql	2020-04-18 22:58:26.108695188 +0200
@@ -49,8 +49,8 @@
 	#	rlm_sql_iodbc
 	#	rlm_sql_unixodbc
 	#
-	driver = "rlm_sql_null"
-#	driver = "rlm_sql_${dialect}"
+#	driver = "rlm_sql_null"
+	driver = "rlm_sql_${dialect}"
 
 	#
 	#  Driver-specific subsections.  They will only be loaded and
@@ -61,7 +61,7 @@
 	#
 	sqlite {
 		# Path to the sqlite database
-		filename = "/tmp/freeradius.db"
+		filename = "/var/lib/radiusd/freeradius.db"
 
 		# How long to wait for write locks on the database to be
 		# released (in ms) before giving up.
diff -Naur --no-dereference default-config/radiusd.conf config/radiusd.conf
--- default-config/radiusd.conf	1970-01-01 01:00:01.000000000 +0100
+++ config/radiusd.conf	2020-04-18 23:02:54.047402250 +0200
@@ -99,9 +99,7 @@
 certdir = ${confdir}/certs
 cadir   = ${confdir}/certs
 run_dir = ${localstatedir}/run/${name}
-
-# Should likely be ${localstatedir}/lib/radiusd
-db_dir = ${raddbdir}
+db_dir = ${localstatedir}/lib/${name}
 
 #
 # libdir: Where to find the rlm_* modules.
@@ -295,7 +293,7 @@
 	#  The command-line option "-X" over-rides this option, and forces
 	#  logging to go to stdout.
 	#
-	destination = files
+	destination = stdout
 
 	#
 	#  Highlight important messages sent to stderr and stdout.
@@ -335,7 +333,7 @@
 	#
 	#  allowed values: {no, yes}
 	#
-	auth = no
+	auth = yes
 
 	#  Log Access-Accept results to the log file.
 	#
@@ -508,7 +506,7 @@
 	#
 	#  See also raddb/sites-available/status
 	#
-	status_server = yes
+	status_server = no
 
 	#
 	#  allow_vulnerable_openssl: Allow the server to start with
@@ -537,7 +535,7 @@
 #
 #  allowed values: {no, yes}
 #
-proxy_requests  = yes
+proxy_requests  = no
 $INCLUDE proxy.conf
 
 
@@ -576,7 +574,7 @@
 thread pool {
 	#  Number of servers to start initially --- should be a reasonable
 	#  ballpark figure.
-	start_servers = 5
+	start_servers = 2
 
 	#  Limit on the total number of servers running.
 	#
@@ -599,7 +597,7 @@
 	#
 	#  For more information, see 'max_request_time', above.
 	#
-	max_servers = 32
+	max_servers = 3
 
 	#  Server-pool size regulation.  Rather than making you guess
 	#  how many servers you need, FreeRADIUS dynamically adapts to
@@ -613,8 +611,8 @@
 	#  more than max_spare_servers, some of the spares die off.
 	#  The default values are probably OK for most sites.
 	#
-	min_spare_servers = 3
-	max_spare_servers = 10
+	min_spare_servers = 1
+	max_spare_servers = 2
 
 	#  When the server receives a packet, it places it onto an
 	#  internal queue, where the worker threads (configured above)
diff -Naur --no-dereference default-config/sites-available/default config/sites-available/default
--- default-config/sites-available/default	1970-01-01 01:00:01.000000000 +0100
+++ config/sites-available/default	2020-04-17 04:01:08.485281275 +0200
@@ -117,7 +117,7 @@
 	#
 	# ipv4addr = *
 	# ipv6addr = *
-	ipaddr = *
+	ipaddr = 127.0.0.1
 
 	#  Port on which to listen.
 	#  Allowed values are:
@@ -202,7 +202,7 @@
 #  port, too.
 #
 listen {
-	ipaddr = *
+	ipaddr = 127.0.0.1
 #	ipv6addr = ::
 	port = 0
 	type = acct
@@ -244,7 +244,7 @@
 # IPv6 versions of the above - read their full config to understand options
 listen {
 	type = auth
-	ipv6addr = ::	# any.  ::1 == localhost
+	ipv6addr = ::1
 	port = 0
 #	interface = eth0
 #	clients = per_socket_clients
@@ -256,7 +256,7 @@
 }
 
 listen {
-	ipv6addr = ::
+	ipv6addr = ::1
 	port = 0
 	type = acct
 #	interface = eth0
@@ -583,9 +583,9 @@
 	#  The start time is: NOW - delay - session_length
 	#
 
-#	update request {
-#	  	&FreeRADIUS-Acct-Session-Start-Time = "%{expr: %l - %{%{Acct-Session-Time}:-0} - %{%{Acct-Delay-Time}:-0}}"
-#	}
+	update request {
+	  	&FreeRADIUS-Acct-Session-Start-Time = "%{expr: %l - %{%{Acct-Session-Time}:-0} - %{%{Acct-Delay-Time}:-0}}"
+	}
 
 
 	#
@@ -621,13 +621,13 @@
 	#  Create a 'detail'ed log of the packets.
 	#  Note that accounting requests which are proxied
 	#  are also logged in the detail file.
-	detail
+#	detail
 #	daily
 
 	#  Update the wtmp file
 	#
 	#  If you don't use "radlast", you can delete this line.
-	unix
+	#unix
 
 	#
 	#  For Simultaneous-Use tracking.
@@ -644,7 +644,7 @@
 	#  Log traffic to an SQL database.
 	#
 	#  See "Accounting queries" in mods-available/sql
-	-sql
+	sql
 
 	#
 	#  If you receive stop packets with zero session length,
@@ -664,7 +664,7 @@
 #	pgsql-voip
 
 	# For Exec-Program and Exec-Program-Wait
-	exec
+	#exec
 
 	#  Filter attributes from the accounting response.
 	attr_filter.accounting_response
@@ -686,7 +686,7 @@
 
 	#
 	#  See "Simultaneous Use Checking Queries" in mods-available/sql
-#	sql
+	sql
 }
 
 
@@ -741,7 +741,7 @@
 	#
 	#  The "sql_session_start" policy is defined in
 	#  raddb/policy.d/accounting.  See that file for more details.
-#	sql_session_start
+	sql_session_start
 
 	#
 	#  If you want to have a log of authentication replies,
@@ -753,7 +753,7 @@
 	#  After authenticating the user, do another SQL query.
 	#
 	#  See "Authentication Logging Queries" in mods-available/sql
-	-sql
+	sql
 
 	#
 	#  Un-comment the following if you want to modify the user's object
diff -Naur --no-dereference default-config/sites-available/inner-tunnel config/sites-available/inner-tunnel
--- default-config/sites-available/inner-tunnel	1970-01-01 01:00:01.000000000 +0100
+++ config/sites-available/inner-tunnel	2020-04-17 03:49:30.410903826 +0200
@@ -259,11 +259,11 @@
 #  or rlm_sql module can handle this.
 #  The rlm_sql module is *much* faster
 session {
-	radutmp
+#	radutmp
 
 	#
 	#  See "Simultaneous Use Checking Queries" in sql.conf
-#	sql
+	sql
 }
 
 
@@ -304,7 +304,7 @@
 	#  After authenticating the user, do another SQL query.
 	#
 	#  See "Authentication Logging Queries" in sql.conf
-	-sql
+	sql
 
 	#
 	#  Un-comment the following if you have set
@@ -369,7 +369,7 @@
 	#
 	Post-Auth-Type REJECT {
 		# log failed authentications in SQL, too.
-		-sql
+		sql
 		attr_filter.access_reject
 
 		#
