diff --git a/nginx_ldap_auth/app/main.py b/nginx_ldap_auth/app/main.py
index 75ba166..cfd1030 100644
--- a/nginx_ldap_auth/app/main.py
+++ b/nginx_ldap_auth/app/main.py
@@ -287,6 +287,9 @@ async def logout(request: Request) -> RedirectResponse:
     if username := request.session.get("username"):
         await kill_session(request)
         _logger.info("auth.logout", username=username)
+    else:
+        session_id = get_session_handler(request).session_id
+        _logger.info("auth.logout.noop", session_id=session_id, headers=request.headers, cookies=request.cookies)
     return RedirectResponse(url="/auth/login?service=/")
 
 
