diff --git a/nginx_ldap_auth/app/main.py b/nginx_ldap_auth/app/main.py
index 4ac98fc..75ba166 100644
--- a/nginx_ldap_auth/app/main.py
+++ b/nginx_ldap_auth/app/main.py
@@ -14,6 +14,7 @@ from starsessions import (
     SessionStore,
     get_session_handler,
     load_session,
+    generate_session_id,
 )
 from starsessions.stores.redis import RedisStore
 
@@ -63,7 +64,7 @@ class CsrfSettings(BaseSettings):
     #: The secret key to use for CSRF tokens
     secret_key: str = Field(validation_alias="CSRF_SECRET_KEY")
     #: We'll set the SameSite attribute on our CSRF cookies to this value
-    cookie_samesite: str = "lax"
+    cookie_samesite: str = "strict"
     #: Set our CSRF cookie to be secure
     cookie_secure: bool = True
     #: Set the maximum age of our CSRF cookie to 5 minutes
@@ -182,6 +183,8 @@ async def login(
         Otherwise, a rendered login page.
 
     """
+    if request.url.query.startswith("https://") or request.url.query.startswith("/"):
+      service = request.url.query
     csrf_token, signed_token = csrf_protect.generate_csrf_tokens()
     auth_realm = request.headers.get("x-auth-realm", settings.auth_realm)
     _logger = get_logger(request)
@@ -255,6 +258,7 @@ async def login_handler(
     await form.load_data()
     if await form.is_valid():
         await load_session(request)
+        get_session_handler(request).session_id = generate_session_id() + ":" + form.username
         request.session["username"] = form.username
         response: HTMLResponse | RedirectResponse = RedirectResponse(
             url=form.service, status_code=status.HTTP_302_FOUND
