diff --git a/hosts/c3pbvm/c3pb/letsmeet/pkgs/node-packages-app.nix b/hosts/c3pbvm/c3pb/letsmeet/pkgs/node-packages-app.nix
index f87e888..0778ebb 100644
--- a/hosts/c3pbvm/c3pb/letsmeet/pkgs/node-packages-app.nix
+++ b/hosts/c3pbvm/c3pb/letsmeet/pkgs/node-packages-app.nix
@@ -2,3 +2,3 @@
 
-{nodeEnv, fetchurl, fetchgit, nix-gitignore, stdenv, lib, globalBuildInputs ? []}:
+{nodeEnv, fetchurl, fetchgit, edumeetSrc, nix-gitignore, stdenv, lib, globalBuildInputs ? []}:
 
@@ -14412,3 +14412,3 @@ let
     version = "3.4.0";
-    src = ./.;
+    src = "${edumeetSrc}/app";
     dependencies = [
@@ -14884,3 +14884,7 @@ let
       sources."buffer-xor-1.0.3"
-      sources."bufferutil-4.0.1"
+      (sources."bufferutil-4.0.1" // {
+        dependencies = [
+          sources."node-gyp-build-3.7.0"
+        ];
+      })
       sources."builtin-status-codes-3.0.0"
@@ -16761,5 +16756,9 @@ let
       sources."ut_pex-2.0.0"
-      sources."utf-8-validate-5.0.2"
+      (sources."utf-8-validate-5.0.2" // {
+        dependencies = [
+          sources."node-gyp-build-3.7.0"
+        ];
+      })
       (sources."util-0.10.3" // {
         dependencies = [
           sources."inherits-2.0.1"
