--- a/src/bootstrap/doc.rs	2021-01-16 01:29:28.114782019 +0100
+++ b/src/bootstrap/doc.rs	2021-01-16 01:27:40.253674431 +0100
@@ -452,8 +452,8 @@
             compile::std_cargo(builder, target, compiler.stage, &mut cargo);
 
             cargo
-                .arg("-p")
-                .arg(package)
+                //.arg("-p")
+                //.arg(package)
                 .arg("-Zskip-rustdoc-fingerprint")
                 .arg("--")
                 .arg("--markdown-css")
